serving/src/main/java/feast/serving/service/RedisServingService.java

```diff
@@ -198,22 +195,31 @@ private void sendAndProcessMultiGet(
     List<byte[]> jedisResps = sendMultiGet(redisKeys);
     long startTime = System.currentTimeMillis();
     try (Scope scope = tracer.buildSpan("Redis-processResponse").startActive(true)) {
-      String featureSetId =
-          String.format("%s:%d", featureSetRequest.getName(), featureSetRequest.getVersion());
+      FeatureSetSpec spec = featureSetRequest.getSpec();
 
       Map<String, Value> nullValues =
-          featureSetRequest.getFeatureNamesList().stream()
+          featureSetRequest.getFeatureReferences().stream()
               .collect(
                   Collectors.toMap(
-                      name -> featureSetId + ":" + name, name -> Value.newBuilder().build()));
+                      fr -> generateFeastId(fr.getProject(), fr.getName(), fr.getVersion()),
+                      fr -> Value.newBuilder().build()));
 
       for (int i = 0; i < jedisResps.size(); i++) {
         EntityRow entityRow = entityRows.get(i);
         Map<String, Value> featureValues = featureValuesMap.get(entityRow);
 
         byte[] jedisResponse = jedisResps.get(i);
         if (jedisResponse == null) {
-          missingKeyCount.labels(featureSetRequest.getName()).inc();
+          featureSetRequest
+              .getFeatureReferences()
+              .parallelStream()
+              .forEach(
+                  fr ->
+                      missingKeyCount
+                          .labels(
+                              spec.getProject(),
+                              String.format("%s:%d", fr.getName(), fr.getVersion()))
```

Comment:
Is this purely for backward compatibility?

Reply:
I'm logging missing key counts for each feature:version. Should I separate the feature and version?

Reply:
No it's ok